src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,163 @@
+package duke;
+
+import static java.lang.Integer.parseInt;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+/**
+ * Processes the string the user put in.
+ */
+public class Parser {
+    /**
+     * Returns the type of task.
+     * @param response the string the user put in.
+     * @return type of task.
+     */
+    public static String parseCommandType(String response) throws DukeException {
+        if (Objects.equals(response, "bye")) {
+            return "bye";
+        } else if (Objects.equals(response, "list")) {
+            return "list";
+        } else if (response.startsWith("delete")) {
+            return "delete";
+        } else if (response.startsWith("mark")) {
+            return "mark";
+        } else if (response.startsWith("todo") || response.startsWith("deadline") || response.startsWith("event")) {
+            return "task";
+        } else if (response.startsWith("find")) {
+            return "find";
+        } else {
+            return "invalid";
+        }
+    }
+    /**
+     * Returns the number associated with a delete or mark command.
+     * number is reduced by 1 since user will write with 1-indexing while
+     * taskList is 0-indexed.
+     * Not to be used for other types of command.
+     * @param response the string the user put in.
+     * @return task number.
+     */
+    public static int taskNumber(String response) {
+        String[] array = response.split(" ");
+        String lastVal = array[array.length - 1];
+        return parseInt(lastVal) - 1;
+    }
+
+    /**
+     * Returns the task given in a command.
+     * @param response the string the user put in.
+     * @return the task.
+     * @throws DukeException if incomplete task or not a task.
+     */
+    public static Task newTask(String response) throws DukeException {
```

Comment:
Consider using helper methods at commands for each task type to improve code readability.